dashboard/client/src/components/NewEventTable.tsx

```diff
@@ -0,0 +1,335 @@
+import {
+  Box,
+  InputAdornment,
+  LinearProgress,
+  makeStyles,
+  Paper,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  TextField,
+  TextFieldProps,
+  Tooltip,
+} from "@material-ui/core";
+import { SearchOutlined } from "@material-ui/icons";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+import Pagination from "@material-ui/lab/Pagination";
+import dayjs from "dayjs";
+import React, { useEffect, useState } from "react";
+import { Align, Event, Filters } from "../type/event";
+
+import { useFilter } from "../util/hook";
+import { SeverityLevel } from "./event";
+import { StatusChip } from "./StatusChip";
+import { useEvents } from "./useEvents";
+
+type EventTableProps = {
+  defaultSeverityLevels?: SeverityLevel[];
+  entityName?: string;
+  entityId?: string; // It could be a specific or "*" to represent all entities
+};
+
+const appendToParams = (
+  params: URLSearchParams,
+  key: string,
+  value: string | string[],
+) => {
+  if (Array.isArray(value)) {
+    value.forEach((val) =>
+      params.append(encodeURIComponent(key), encodeURIComponent(val)),
+    );
+  } else {
+    params.append(encodeURIComponent(key), encodeURIComponent(value));
+  }
+};
```

Comment:
can you move this to a util file and write a unit test for it?

Reply:
Sounds good!